src/app/login/login.component.ts

```diff
@@ -43,4 +43,11 @@ export class LoginComponent implements OnInit {
     );
   }
 
+  checkAuthSession() {
+    return this.couchService.get('_session')

```

Comment:
This does not work as you intended: it will simply return the Observable which evaluates to `true`.  Please remember that callbacks in asynchronous functions (promises, observables, etc) don't push their return value up to a surrounding function.